tests/bson/bson-fromPHP_error-004.phpt

```diff
@@ -0,0 +1,47 @@
+--TEST--
+BSON\fromPHP(): PHP documents with circular references
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/tools.php';
+
+echo "\nTesting packed array with circular reference\n";
+
+echo throws(function() {
+    $document = ['x' => 1, 'y' => []];
+    $document['y'][] = &$document['y'];
+    fromPHP($document);
+}, 'MongoDB\Driver\Exception\UnexpectedValueException'), "\n";
+
+echo "\nTesting associative array with circular reference\n";
+
+echo throws(function() {
+    $document = ['x' => 1, 'y' => []];
+    $document['y']['z'] = &$document['y'];
+    fromPHP($document);
+}, 'MongoDB\Driver\Exception\UnexpectedValueException'), "\n";
+
+echo "\nTesting object with circular reference\n";
+
+echo throws(function() {
+    $document = (object) ['x' => 1, 'y' => (object) []];
+    $document->y->z = &$document->y;
+    fromPHP($document);
+}, 'MongoDB\Driver\Exception\UnexpectedValueException'), "\n";
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECTF--
+Testing packed array with circular reference
+OK: Got MongoDB\Driver\Exception\UnexpectedValueException
+Detected recursion for fieldname "0"
```

Comment:
Let me know if you'd prefer changing "fieldname" to "array index" for clarity. Unfortunately, we aren't tracking the full property path, but that may be something we can improve in [PHPC-922](https://jira.mongodb.org/browse/PHPC-922) (I've left a note).